library/ssl_cli.c

```diff
@@ -4361,7 +4361,14 @@ int mbedtls_ssl_handshake_client_step( mbedtls_ssl_context *ssl )
            ret = mbedtls_ssl_parse_delayed_certificate_verify( ssl, authmode,
                                    ssl->session_negotiate->peer_cert, rs_ctx );
            if( ret != 0 )
+           {
```

Comment:
it seems we should check this error twice without if inside if. It should be 2 separate if conditions and enoght that one of the is error we want to return error

Reply:
Updated.